e3sm_diags/driver/utils/regrid.py

```diff
@@ -526,7 +526,8 @@ def _hybrid_to_plevs(
             method="log",
             target_data=pressure_coords,
         )
-
+    # Keep units unchanged.
+    result[var_key].attrs["units"] = ds[var_key].units
```

Comment:
I'm copied this fix over to PR #774 in [`ecc2102` (#774)](https://github.com/E3SM-Project/e3sm_diags/pull/774/commits/ecc2102d75213809a24eae5967323bdbf10b6b56) since that PR will be merged shortly.